### PR TITLE
feat: Add permissions mapping to reclique_sso

### PR DESCRIPTION
**Pull request by froboy**
Adds permissions mapping to gc_auth_reclique_sso. This change attempts to keep the existing behavior the same but add a number of options to increase flexibility for different ReClique setups. It also cleans up some changes from #2.

The ReClique SSO configuration now has a number of additional fields, all of which default to their prior behavior but provide options for flexible permissions mapping.

Identity_Provider___YMCA_of_Metropolitan_Dallas

## Steps to test:

- [ ] Enable openy_gc_auth_reclique_sso
- [ ] Configure it, observe the settings form saves correctly.
- [ ] Ensure folks can log in and receive the proper roles.

## Quality checks:

Please check these boxes to confirm this PR covers the following cases:

- Maintaining our upgrade path is essential. Check one or the other:
  - [ ] This PR  provides updates via `hook_update_N` or other means.
  - [ ] No updates are necessary for this change.
- Front end fixes should be tested against all of the Open Y Themes.
  - [ ] Tested against Carnation
  - [ ] Tested against Lily
  - [ ] Tested against Rose
  - [ ] This change does not contain front-end fixes.
- [ ] I have flagged this PR "Needs Review" or pinged the VY devs/QA
  team in Slack

### CODE REVIEW

**Review by anpolimus — APPROVED** [review]

**Comment by anpolimus** [comment]
@froboy only upgrade path is missing, but I never saw this module used. Probably we could skip it.

**Comment by froboy** [comment]
@anpolimus I've found that with new fields the `defaultConfiguration` method is enough to fill in the gaps, and since I'm only adding and not changing anything i think that should be enough.

File: modules/openy_gc_auth/modules/openy_gc_auth_reclique_sso/config/install/openy_gc_auth.provider.reclique_sso.yml [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
-error_not_found: 'That user was not found.'
 error_access_denied: 'That user does not have access to Virtual Y.'
-error_invalid: 'Something went wrong.'
 error_accompanying_message: 'Please contact us if you have any questions.'
+error_invalid: 'Something went wrong.'
+error_not_found: 'That user was not found.'
 login_mode: 'present_login_button'
+membership_field: 'PackageName'
+permissions_mapping: ''
+require_active: 1
```
Comment (podarok) [inline modules/openy_gc_auth/modules/openy_gc_auth_reclique_sso/config/install/openy_gc_auth.provider.reclique_sso.yml:11]: We should have hook_update_N for this config to be updated

**Review by podarok — CHANGES_REQUESTED** [review]
Let's have update path for config

File: modules/openy_gc_auth/modules/openy_gc_auth_reclique_sso/config/install/openy_gc_auth.provider.reclique_sso.yml (hunk 1 above)
Comment (froboy) [reply modules/openy_gc_auth/modules/openy_gc_auth_reclique_sso/config/install/openy_gc_auth.provider.reclique_sso.yml:11]: @podarok these are all new fields and have default values https://github.com/YCloudYUSA/yusaopeny_gated_content/pull/4/files#diff-f494604cd8afa5f9c202ee02db5e7cfb894c493618f019fe30baf837c12ec481R102
so I believe from prior conversations they don't need an update hook.

**Comment by froboy** [comment]
@podarok see https://github.com/YCloudYUSA/yusaopeny_gated_content/pull/4#discussion_r984915290. I don't think update hook is necessary here.

**Pull request merged**